.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,98 @@
+name: Docker
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+on:
+  schedule:
+    - cron: '31 6 * * *'
+  push:
+    branches: [ "main" ]
+    # Publish semver tags as releases.
+    tags: [ 'v*.*.*' ]
+  pull_request:
+    branches: [ "main" ]
```

Comment:
Realistically, we should not be running this for PRs, opens the possibility for malware to be added without needing a merge.

Reply:
It runs on PR's to test docker image building, but it does not push, see this comment someone over at gh made 

`
      # Build and push Docker image with Buildx (don't push on PR)
      # https://github.com/docker/build-push-action
`

Reply:
see this as well
```
        uses: docker/build-push-action@0565240e2d4ab88bba5387d719585280857ece09 # v5.0.0
        with:
          context: .
          push: ${{ github.event_name != 'pull_request' }}
``` 

signature tooling also gets skipped on PRs
```
      # Install the cosign tool except on PR
      # https://github.com/sigstore/cosign-installer
      - name: Install cosign
        if: github.event_name != 'pull_request'
        uses: sigstore/cosign-installer@e1523de7571e31dbe865fd2e80c5c7c23ae71eb4 #v3.4.0
        with:
          cosign-release: 'v2.2.3'

```

```
      # Sign the resulting Docker image digest except on PRs.
      # This will only write to the public Rekor transparency log when the Docker
      # repository is public to avoid leaking data.  If you would like to publish
      # transparency data even for private images, pass --force to cosign below.
      # https://github.com/sigstore/cosign
      - name: Sign the published Docker image
        if: ${{ github.event_name != 'pull_request' }}
        env:
          # https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-an-intermediate-environment-variable
          TAGS: ${{ steps.meta.outputs.tags }}
          DIGEST: ${{ steps.build-and-push.outputs.digest }}
        # This step uses the identity token to provision an ephemeral certificate
        # against the sigstore community Fulcio instance.
        run: echo "${TAGS}" | xargs -I {} cosign sign --yes {}@${DIGEST}
```

Reply:
also when you ran the docker build workflow on this PR, it didn't push an image to the container registry, as https://github.com/orgs/LBPUnion/packages?repo_name=ProjectLighthouse is empty

Reply:
Is this ok or are there any changes to this you want? @sudokoko

Reply:
Let me check with the team to see if this is something that we'd be fine with including, I'll get back to you on this after I do a more thorough review

Reply:
If there are any edits that need to be made let me know

Reply:
I'm just going to push this aside for a bit, me personally I don't see any problems with this as long as it doesn't publish on PR.